src/Microsoft.ML.Data/DataView/Transposer.cs

```diff
@@ -1204,13 +1140,11 @@ public ColumnSplitter(IDataView view, int col, int[] lims)
                         for (int c = 1; c < _lims.Length; ++c)
                             _types[c] = new VectorType(type.ItemType, _lims[c] - _lims[c - 1]);
 
-                        AsSchema = Schema.Create(this);
-                    }
-
-                    public override ColumnType GetColumnType(int col)
-                    {
-                        Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-                        return _types[col];
+                        var selectedColumn = _view.Schema[col];
+                        var schemaBuilder = new SchemaBuilder();
+                        for (int c = 0; c < _lims.Length; ++c)
+                            schemaBuilder.AddColumn(selectedColumn.Name, _types[c]);
```

Comment:
> schemaBuilder.AddColumn(selectedColumn.Name, _types[c]); [](start = 27, length = 57)

double-checking that omitting metadata is deliberate.

Reply:
It is reasonable because splitting a vector-valued column breaks most metadata. It's why previously `Splitter` was a `NoMetadataSchema`.

---
In reply to: [246463230](https://github.com/dotnet/machinelearning/pull/2080#discussion_r246463230) [](ancestors = 246463230)